lib/stdlib/test/begin_maybe_SUITE.erl

```diff
@@ -229,6 +231,70 @@ end of
             end
     end.
 
+begin6(_) ->
+    ?assertEqual(maybe6(), case6()).
+
+maybe6() ->
+    begin
+        X = 2+2,
+        {ok, 4} <- {error, 5},
+        Z = X*2,
+        {error, expected} <- {ok, 3},
+        id(Z),
+        X+Z
+    cond
+        {error, _} -> a;
+        {ok, _} -> b
+    end.
+
+case6() ->
+    case
+        begin
+            X = 2+2,
+            case {error, 5} of
+                {ok, 4} ->
+                    Z = X*2,
+                    case {ok, 3} of
+                        {error, expected} ->
+                            id(Z),
+                            {successful_begin, X+Z};
+                        Other ->
+                            {begin_fail_branch, Other}
+                    end;
+                Other ->
+                    {begin_fail_branch, Other}
+            end
+        end
+    of
+        {successful_begin, V} -> V;
+        {begin_fail_branch, Err} ->
+            case Err of
+                {error, _} -> a;
+                {ok, _} -> b
+            end
+    end.
+
+begin7(_) ->
+    ?assertEqual(maybe7(), case7()).
+
+maybe7() ->
+    begin
+        3
+    cond
+        {error, _} -> a;
+        {ok, _} -> b
+    end.
```

Comment:
This new form has a little thing where to skip on process, the `cond ... end` is dropped and causes no warning for unused error handling.